### DISCUSSION TITLE
en/lc/648/

### DISCUSSION THREAD

**Discussion by giscus** [doocs/leetcode#3072] [Announcements]
# en/lc/648/

LeetCode solutions in any programming language

https://leetcode.doocs.org/en/lc/648/

**Comment by rain84** [comment]
Interested, why `Array(26)` chosen for  the  initialisation of the every layer of the trie?
(`this.children = new Array<Trie>(26);)`

I guess, usual hash-map `{}` will be better by the usage of space.
And also there will be no need to convert every next character into index with a `.charCodeAt(0)`

**Reply by klever34** [reply]
Well, yeah a hash map only stores the characters that are actually used, potentially saving space if the trie is sparse (i.e., not all 26 children are used at every node).

**Reply by yanglbme** [reply]
Hello, based on my experience with solving LeetCode algorithm problems, for some programming languages such as Java and C++, allocating a static array is more efficient than using a hash table. Therefore, I tend to use a static array instead of a hash table.
